test/metadata.jl

```diff
@@ -470,6 +470,59 @@ end
     @test check_allnotemetadata(df)
 end
 
+@testset "metadata default" begin
+    df = DataFrame(a=1)
+    metadata!(df, "x", "y")
+    colmetadata!(df, :a, "p", "q")
+    @test metadata(df, "x", style=true) == ("y", :default)
+    @test colmetadata(df, :a, "p", style=true) == ("q", :default)
+    df = eachcol(DataFrame(a=1))
+    metadata!(df, "x", "y")
+    colmetadata!(df, :a, "p", "q")
+    @test metadata(df, "x", style=true) == ("y", :default)
+    @test colmetadata(df, :a, "p", style=true) == ("q", :default)
+    df = view(DataFrame(a=1), :, :)
+    @test_throws ArgumentError metadata!(df, "x", "y")
+    @test_throws ArgumentError colmetadata!(df, :a, "p", "q")
+end
+
+@testset "fallback definitions of metadata and colmetadata" begin
```

Comment:
Don't we want to define methods that return internal fields instead of relying on the fallbacks? Or only do that in the next minor release?

Reply:
The reason why I wanted to use fallbacks is the following:
* in most cases users will ask for `style=false`, in which case we need to use fallback anyway (to strip style).
* if `style=true` is passed indeed we could return internal dictionary for table-level metadata and column-level metadata when `col` is passed, but not `colmetadata(df)` (the reason is that this dictionary has integers as keys; and it has integers as keys so that functions like `rename!` can have a simple implementation).

Now, I thought that users will not call `metadata(df)` or `colmetadata(df)` inside hot-loops (as these functions are mostly designed to be called once), so I thought it would be safer to make a copy even in cases when we could avoid it (as it will not affect performance much anyway).